src/github-graphql-base.js

```diff
@@ -0,0 +1,600 @@
+
+    // sleep time (sec) before timeout retry
+    this.sleepTime = sleepTime;
+
+    // maximum number of warnings before stop fetching
+    this.warningThreshold = warningThreshold;
+  }
+
+  repo(repoOwner, repoName) {
+    this.repoOwner = repoOwner;
+    this.repoName = repoName;
+    if (DEBUG) {
+      console.log('repoOwner set:', repoOwner, 'repoName set:', repoName);
+    }
+    return this;
+  }
+
+  // First type of queries: ISSUES
+  issues(config) {
+    const { sort, direction, earliestDate } = config || {};
+    // There are three types of order
+    // https://developer.github.com/v4/enum/issueorderfield/
+    // COMMENTS, CREATED_AT (default), and UPDATED_AT
+    this.orderBy = {
+      field: sort || 'CREATED_AT',
+      direction: direction || 'ASC'
+    };
+    this._fetch = this._fetchIssues;
+    this.earliestDate = earliestDate;
+    return this;
+  }
+
+  // Second type of queries: PULLREQUESTS
+  pullRequests(config) {
+    const { sort, direction, earliestDate,
+      ignoreAuthor, ignoreContent } = config || {};
+    // order of pull requests and issues follow the same rule
+    // see comment within `issues` method
+    this.orderBy = {
+      field: sort || 'CREATED_AT',
+      direction: direction || 'ASC'
+    };
+    this._fetch = this._fetchPullRequests;
+    this.earliestDate = earliestDate;
+
+    // review will be ignored if it matches any string specified here
+    // it can be a literal or a regular expression
+    // delimiter: space
+    if (ignoreAuthor || ignoreAuthor === '') {
+      this.ignoreAuthor = ignoreAuthor;
+    } else {
+      // default value
+      this.ignoreAuthor = 'gitmate-bot rultor';
+    }
+
+    // review will be ignored if it's written by any author specified here
+    // delimiter: space
+    if (ignoreContent || ignoreContent === '') {
+      this.ignoreContent = ignoreContent;
+    } else {
+      // default value
+      this.ignoreContent = '@gitmate-bot @rultor /^(unack|ack)/g';
+    }
+
+    return this;
+  }
+
+  // Third type of queries: LABELS
+  labels() {
+    this._fetch = this._fetchLabels;
+    return this;
+  }
+
+  // Fourth type of queries: REACTIONS
+  reactions(config) {
+    const { pr_number, reviews_count,
+      comments_count, discussions_per_review } = config || {};
+    this._fetch = this._fetchReactions;
+    this.prNumber = pr_number;
+    // number of comments to fetch per pull request
+    this.reviewsCount = reviews_count || 20;
+    // number of discussions to fetch per review
+    this.discussionsPerReview = discussions_per_review || 10;
+    // number of comments to fetch per pull request
+    this.commentsCount = comments_count || 20;
+    return this;
+  }
+
+  async fetchAll(config) {
+    const { per_page } = config || {};
+    this.perPage = per_page || 100;
+    this.cursor = null;
+    this.pageCount = 0;
+    this.fetchedData = null;
+
+    if (this._fetch === this._fetchLabels
+      || this._fetch === this._fetchReactions) {
```

Comment:
this is my python brain kicking in ....

the `|| ..` here should be indented more so that it doesnt align with the `console.log(..)` on the next line.